rmgpy/rmg/react.py

```diff
@@ -118,59 +120,73 @@ def react_all(core_spc_list, numOldCoreSpecies, unimolecularReact, bimolecularRe
+                            spc_tuples.append(((core_spc_list[i], core_spc_list[j]), family))
+                        else:
+                            fam_leftovers.append(family)
+            if fam_leftovers:
+                spc_tuples.append(((core_spc_list[i], core_spc_list[j]), fam_leftovers))
 
     if trimolecularReact is not None:
         for i in xrange(numOldCoreSpecies):
             for j in xrange(i, numOldCoreSpecies):
                 for k in xrange(j, numOldCoreSpecies):
-                    # Find reactions involving the species that are trimolecular.
-                    if trimolecularReact[i, j, k]:
-                        if core_spc_list[i].reactive and core_spc_list[j].reactive and core_spc_list[k].reactive:
-                            spc_tuples.append((core_spc_list[i], core_spc_list[j], core_spc_list[k]))
-
-    if procnum == 1:
-        # React all families like normal (provide empty argument for only_families)
-        spc_fam_tuples = zip(spc_tuples)
-    else:
-        # Identify and split families that are prone to generate many reactions into sublists.
-        family_list = getDB('kinetics').families.keys()
-        major_families = [
-            'H_Abstraction', 'R_Recombination', 'Intra_Disproportionation', 'Intra_RH_Add_Endocyclic',
-            'Singlet_Carbene_Intra_Disproportionation', 'Intra_ene_reaction', 'Disproportionation',
-            '1,4_Linear_birad_scission', 'R_Addition_MultipleBond', '2+2_cycloaddition_Cd', 'Diels_alder_addition',
-            'Intra_RH_Add_Exocyclic', 'Intra_Retro_Diels_alder_bicyclic', 'Intra_2+2_cycloaddition_Cd',
-            'Birad_recombination', 'Intra_Diels_alder_monocyclic', '1,4_Cyclic_birad_scission', '1,2_Insertion_carbene',
-        ]
-
-        split_list = []
-        leftovers = []
-        for fam in family_list:
-            if fam in major_families:
-                split_list.append([fam])
-            else:
-                leftovers.append(fam)
-        split_list.append(leftovers)
-
-        # Only employ family splitting for reactants that have a larger number than min_atoms
-        min_atoms = 10
-        spc_fam_tuples = []
-        for i, spc_tuple in enumerate(spc_tuples):
-            if any([len(spc.molecule[0].atoms) > min_atoms for spc in spc_tuple]):
-                for item in split_list:
-                    spc_fam_tuples.append((spc_tuple, item))
-            else:
-                spc_fam_tuples.append((spc_tuple, ))
-
-    return react(spc_fam_tuples, procnum), [fam_tuple[0] for fam_tuple in spc_fam_tuples]
+                    fam_leftovers = []
+                    for l, family in enumerate(family_names):
+                        # Find reactions involving the species that are trimolecular
+                        if (len(families[k].forwardTemplate.reactants) == 3 or len(
+                                families[k].forwardTemplate.products) == 3):
```

Comment:
Could you put the `len(` on the second line?

Reply:
´len(´on second line now.